apiserver/facades/client/applicationoffers/applicationoffers.go

```diff
@@ -225,11 +261,14 @@ func (api *OffersAPIv5) modifyOneOfferAccess(ctx context.Context, user names.Use
 		if err != nil {
 			return apiservererrors.ErrPerm
 		}
-		access, err := backend.GetOfferAccess(offer.OfferUUID, user)
-		if err != nil && !errors.Is(err, errors.NotFound) {
+		accessLevel, err := api.accessService.ReadUserAccessLevelForTarget(ctx, coreuser.NameFromTag(user), permission.ID{
+			ObjectType: permission.Offer,
+			Key:        offer.OfferUUID,
+		})
+		if err != nil && !errors.Is(err, accesserrors.AccessNotFound) {
```

Comment:
I had a discussion with Joe that we would move this check out of the method last week. The idea is that its for consistency. I've [got a task](https://warthogs.atlassian.net/browse/JUJU-6563) to put move the check in the apiserver rather than have it in the state.

I was intending to leave this in until then.